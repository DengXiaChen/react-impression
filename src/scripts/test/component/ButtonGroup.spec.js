import React from 'react';
import { shallow } from 'enzyme';
import { ButtonGroup, Button } from '../../components/impression';

const setup = () => {
    const props = {
        theme: 'secondary',
        size: 'lg',
        activeKey: 'right',
        onSelect: jest.fn(),
    };

    const wrapper = shallow(
        <ButtonGroup {...props}>
            <Button eventKey="left">Left</Button>
            <Button eventKey="middle">Middle</Button>
            <Button eventKey="right">Right</Button>
        </ButtonGroup>
    );

    return {
        props,
        wrapper,
    };
};

describe('ButtonGroup', () => {
    const { props, wrapper } = setup();

    it('render', () => {
        expect(wrapper.find('.btn-group').exists()).toBe(true);
    });

    it('size', () => {
        expect(wrapper.find('.btn-group-lg').exists()).toBe(true);
    });

    it('theme', () => {
        expect(wrapper.find('Button')).toHaveLength(3);
    });

    it('activeKey', () => {
        expect(wrapper.state().activeKey).toBe('right');
    });

    it('onSelect', () => {
        wrapper.find('Button').at(1).simulate('click');
        expect(props.onSelect).toHaveBeenCalledWith('middle', undefined);
        expect(wrapper.state().activeKey).toBe('middle');
    });

    it('ui', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
