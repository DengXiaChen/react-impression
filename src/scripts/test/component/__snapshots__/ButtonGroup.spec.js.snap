// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup ui 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="btn-group btn-group-lg"
>
    <Button
        eventKey="left"
        onClick={[Function]}
        outline={true}
        theme="secondary"
    >
        Left
    </Button>
    <Button
        eventKey="middle"
        onClick={[Function]}
        outline={false}
        theme="secondary"
    >
        Middle
    </Button>
    <Button
        eventKey="right"
        onClick={[Function]}
        outline={true}
        theme="secondary"
    >
        Right
    </Button>
</div>,
  "nodes": Array [
    <div
      className="btn-group btn-group-lg"
>
      <Button
            eventKey="left"
            onClick={[Function]}
            outline={true}
            theme="secondary"
      >
            Left
      </Button>
      <Button
            eventKey="middle"
            onClick={[Function]}
            outline={false}
            theme="secondary"
      >
            Middle
      </Button>
      <Button
            eventKey="right"
            onClick={[Function]}
            outline={true}
            theme="secondary"
      >
            Right
      </Button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 1,
      "_context": Object {},
      "_currentElement": <ButtonGroup
        activeKey="right"
        onSelect={[Function]}
        size="lg"
        theme="secondary"
>
        <Button
                eventKey="left"
                theme="primary"
        >
                Left
        </Button>
        <Button
                eventKey="middle"
                theme="primary"
        >
                Middle
        </Button>
        <Button
                eventKey="right"
                theme="primary"
        >
                Right
        </Button>
</ButtonGroup>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ButtonGroup {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "activeKey": "right",
          "children": Array [
            <Button
              eventKey="left"
              theme="primary"
>
              Left
</Button>,
            <Button
              eventKey="middle"
              theme="primary"
>
              Middle
</Button>,
            <Button
              eventKey="right"
              theme="primary"
>
              Right
</Button>,
          ],
          "onSelect": [Function],
          "size": "lg",
          "theme": "secondary",
        },
        "refs": Object {},
        "state": Object {
          "activeKey": "middle",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="btn-group btn-group-lg"
>
          <Button
                    eventKey="left"
                    onClick={[Function]}
                    outline={true}
                    theme="secondary"
          >
                    Left
          </Button>
          <Button
                    eventKey="middle"
                    onClick={[Function]}
                    outline={false}
                    theme="secondary"
          >
                    Middle
          </Button>
          <Button
                    eventKey="right"
                    onClick={[Function]}
                    outline={true}
                    theme="secondary"
          >
                    Right
          </Button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="btn-group btn-group-lg"
>
          <Button
                    eventKey="left"
                    onClick={[Function]}
                    outline={true}
                    theme="secondary"
          >
                    Left
          </Button>
          <Button
                    eventKey="middle"
                    onClick={[Function]}
                    outline={false}
                    theme="secondary"
          >
                    Middle
          </Button>
          <Button
                    eventKey="right"
                    onClick={[Function]}
                    outline={true}
                    theme="secondary"
          >
                    Right
          </Button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ButtonGroup
    activeKey="right"
    onSelect={[Function]}
    size="lg"
    theme="secondary"
>
    <Button
        eventKey="left"
        theme="primary"
    >
        Left
    </Button>
    <Button
        eventKey="middle"
        theme="primary"
    >
        Middle
    </Button>
    <Button
        eventKey="right"
        theme="primary"
    >
        Right
    </Button>
</ButtonGroup>,
}
`;
